### PR TITLE
Added support for deprecated elements in ipp-registrations.xml

### PR DESCRIPTION
**Pull request by HPNavjot**
Addresses issue #130

### CODE REVIEW

File: jipp-core/src/main/java/com/hp/jipp/model/Types.kt [hunk 1]
```diff
@@ -388,6 +414,11 @@ object Types {
      * [RFC8011](http://www.iana.org/go/rfc8011).
      */
     @JvmField val documentFormatDefault = StringType(Tag.mimeMediaType, "document-format-default")
+    /**
+     * "document-format-details" type as defined in:
+     * [PWG5100.7](https://ftp.pwg.org/pub/pwg/candidates/cs-ippjobext20-20190816-5100.7.pdf).
+     */
+    @JvmField val documentFormatDetails = AttributeCollection.Type("document-format-details", DocumentFormatDetails)
```
Comment (HPSudip) [inline jipp-core/src/main/java/com/hp/jipp/model/Types.kt:421]: if we are bringing this code back we need to remove this line 
```
ignored_attributes = [
    # XML Fix (obsolete or maybe just deprecated)
    'document-format-details-supported'
]
```

in that case the -supported will also come.

File: jipp-core/bin/genTypes.py [hunk 2]
```diff
@@ -354,7 +354,7 @@ def parse_attribute(record):
         attr_name = re.sub(' *\(.*\)', '', attr_name)
 
         # Do not delete attributes with extension suffix
-        if not (suffix.group(1) == 'extension'):
+        if not (suffix.group(1) == 'extension' or suffix.group(1) == 'deprecated'):
```
Comment (HPSudip) [inline jipp-core/bin/genTypes.py:357]: Just extend the comment above

**Review by HPSudip — CHANGES_REQUESTED** [review]
Looks good Navjot, just 2 comments.

File: jipp-core/bin/genTypes.py (hunk 2 above)
Comment (HPNavjot) [reply jipp-core/bin/genTypes.py:362]: Done

**Review by HPSudip — APPROVED** [review]

**Pull request merged**